src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -114,6 +139,89 @@ void KevinHallModel::update_risk_factors(RuntimeContext &context) {
     }
 }
 
+void KevinHallModel::initialise_sector(RuntimeContext &context, Person &person) const {
+
+    // Get rural prevalence for age group and sex.
+    double prevalence;
+    if (person.age < 18) {
+        prevalence = rural_prevalence_.at("Under18"_id).at(person.gender);
+    } else {
+        prevalence = rural_prevalence_.at("Over18"_id).at(person.gender);
+    }
+
+    // Sample the person's sector.
+    double rand = context.random().next_double();
+    auto sector = rand < prevalence ? core::Sector::rural : core::Sector::urban;
+    person.sector = sector;
```

Comment:
This is a perfectly fine way of setting `sector` randomly, but I just want to double-check that the model is designed this way on purpose...

Another way to do it would be to say e.g. for every 100 people, we want 60 assigned as rural and 40 as urban, then shuffle the order. That way you would always have the exact proportion as defined in the config file.

Out of interest, how are other attributes (e.g. gender) assigned randomly?

Reply:
I think the stochasticity is important to them, even though I agree that it's uncertain what purpose it is serving in this case. What you suggest is pretty much how they do it for age and sex in `PopulationModule::initialise_population`. But they don't shuffle, so I think you get bit blocks of heterogeneous people. Therefore I think the stochasticity is warranted here, so you don't get all older males living in the city and younger women living in farms. Also, since new people are periodically initialised, you would presumably need to shuffle every iteration, which sounds expensive.

Reply:
Ah I see. If you're doing it every iteration, then it makes more sense. I think I was assuming this was done during initialisation.